src/main/java/org/jasig/portlet/newsreader/dao/HibernateNewsStore.java

```diff
@@ -22,335 +22,233 @@
-        try {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
 
-            getHibernateTemplate().saveOrUpdate(listing);
-            getHibernateTemplate().flush();
+    private SessionFactory sessionFactory;
 
-        } catch (HibernateException ex) {
-            throw convertHibernateAccessException(ex);
-        }
+    public void setSessionFactory(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
     }
 
-    public void storeNewsConfiguration(NewsConfiguration configuration) {
-        try {
-
-            getHibernateTemplate().saveOrUpdate(configuration);
-            getHibernateTemplate().flush();
-
-        } catch (HibernateException ex) {
-            throw convertHibernateAccessException(ex);
-        }
+    private Session currentSession() {
+        return sessionFactory.getCurrentSession();
     }
 
-    public List<NewsConfiguration> getNewsConfigurations(
-            String subscribeId) {
-        try {
-
-            logger.debug("fetching news configurations for " + subscribeId);
-            return (List<NewsConfiguration>) getHibernateTemplate().find(
-                    "from NewsConfiguration config where "
-                            + "subscribeId = ? and displayed = true "
-                            + "order by newsDefinition.name", subscribeId);
-
-        } catch (HibernateException ex) {
-            throw convertHibernateAccessException(ex);
-        }
+    private <T> Query<T> createQuery(String hql) {
+        @SuppressWarnings("unchecked")
+        Query<T> query = sessionFactory.getCurrentSession().createQuery(hql);
+        return query;
     }
 
-    public List<UserDefinedNewsConfiguration> getUserDefinedNewsConfigurations(
-            Long setId, boolean visibleOnly) {
-        try {
-
-            String query = "from NewsConfiguration config where "
-                    + "config.newsSet.id = ? and "
-                    + "config.class = UserDefinedNewsConfiguration "
-                    + "order by newsDefinition.name";
-            if (visibleOnly)
-                query = query.concat(" and visibleOnly = true");
+    public void storeNewsDefinition(NewsDefinition listing) {
+        currentSession().saveOrUpdate(listing);
+        currentSession().flush();
+    }
 
-            return (List<UserDefinedNewsConfiguration>) getHibernateTemplate()
-                    .find(query, setId);
+    public void storeNewsConfiguration(NewsConfiguration configuration) {
+        currentSession().saveOrUpdate(configuration);
+        currentSession().flush();
+    }
 
-        } catch (HibernateException ex) {
-            throw convertHibernateAccessException(ex);
-        }
+    public List<NewsConfiguration> getNewsConfigurations(String subscribeId) {
+        logger.debug("fetching news configurations for " + subscribeId);
+        Query<NewsConfiguration> query = createQuery(
+                "from NewsConfiguration config where "
```

Comment:
I did not intend to make changes to this class, but since I am might as well improve the HQL as well. Thanks @ChristianMurphy for the comment and links.